### PR TITLE
`Development`: Build Docker image. Publish to GitHub Container Registry for develop, tags, and open pull requests.

### PR DESCRIPTION
**Pull request by jpbernius**
This PR sets up @actions to build a `artemis` docker image.
For now, we publish the image as `ghcr.io/ls1intum/artemis` ([View](https://github.com/ls1intum/Artemis/pkgs/container/artemis)) with the following tags:

1. `latest` for the `develop` branch.
2. `x.y.z` for tags/releases.
3. `pr-xxxx` for open pull requests.

PR Tags will be removed once the pr is closed or merged.

### Perks:
- Update `Dockerfile` to OpenJDK 17.
- Add a `docker-compose.yml` for testing.

### CODE REVIEW

**Comment by dfuchss** [comment]
What kind of docker images do you want to create?
We've created docker images for client & server as two images to separate the backend and the actual UI.

E.g., see here: https://github.com/kit-sdq/Artemis-SAML2-Test-Docker

File: src/main/docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        image: ghcr.io/ls1intum/artemis
+        build:
+            context: ../../..
+            dockerfile: src/main/docker/Dockerfile
+        volumes:
+            - ../resources/config/application-dev.yml:/home/artemis/application-dev.yml:ro
+            - ../resources/config/application-artemis.yml:/home/artemis/application-artemis.yml:ro
+        environment:
+            _JAVA_OPTIONS: -Xmx5120m -Xms2560m
+            SPRING_PROFILES_ACTIVE: dev,bamboo,bitbucket,jira,artemis,scheduling,athene,local
```
Comment (Mtze) [inline src/main/docker/docker-compose.yml:13]: Could these profiles cause problems, if the services are note defined in the `application-X.yml`?

**Review by Mtze — COMMENTED** [review]
I love the new GitHub workflow. Very handy to have the image at hand without manually compiling the binary first. 

I tried to start Artemis with the docker-compose file. Unfortunately, the Artemis container crashes with `docker_artemis-app_1 exited with code 137` and no meaningful error message. 
This happens with a local container build as well as with the image available on GitHub. I updated the `application-artemis.yml` with the following changes: 
- configured an `encryption-password`
- removed the `user-management` components  `external` and `ldap` and set the `use-external` to `false`
- removed the complete `version-control` and `continuous-integration` blocks 


To test the GitHub image I updated the docker-compose file to include the `pr-` tag. I also tried to remove the spring profiles for jira, bitbucket and bamboo. This didn't do the trick :( 

I guess I'm "holding it wrong". Any hints to solve the problem?

**Comment by Mtze** [comment]
It works for me now. I updated the resources for docker and clicked apply before starting with the review. Apparently docker reseted the settings after the apply without notice. Updating the resources and restarting docker by hand fixed it for me. 

Now both the local build and GitHub image work fine for me.

**Review by Mtze — APPROVED** [review]

File: src/main/docker/docker-compose.yml (hunk 1 above)
Comment (jpbernius) [reply src/main/docker/docker-compose.yml:13]: yes.
We could add a variable here with these profiles as default.
Or users can create a `docker-compose.overwrite.yml` file.

**Pull request merged**